`frost-core` does not compile without features

How to reproduce:

```text
git checkout frost-core/v1.0.0
cargo check -p frost-core --no-default-features
```

<details><summary>Output</summary>
<p>

```text
error[E0432]: unresolved imports `crate::serialization::Deserialize`, `crate::serialization::Serialize`
   --> frost-core/src/keys.rs:20:21
    |
20  |     serialization::{Deserialize, Serialize},
    |                     ^^^^^^^^^^^  ^^^^^^^^^ no `Serialize` in `serialization`
    |                     |
    |                     no `Deserialize` in `serialization`
    |
note: found an item that was configured out
   --> frost-core/src/serialization.rs:171:18
    |
171 | pub(crate) trait Deserialize<C: Ciphersuite> {
    |                  ^^^^^^^^^^^
    = note: the item is gated behind the `serialization` feature
note: found an item that was configured out
   --> frost-core/src/serialization.rs:165:18
    |
165 | pub(crate) trait Serialize<C: Ciphersuite> {
    |                  ^^^^^^^^^
    = note: the item is gated behind the `serialization` feature

error[E0432]: unresolved imports `crate::serialization::Deserialize`, `crate::serialization::Serialize`
   --> frost-core/src/keys/dkg.rs:53:32
    |
53  |     use crate::serialization::{Deserialize, Serialize};
    |                                ^^^^^^^^^^^  ^^^^^^^^^ no `Serialize` in `serialization`
    |                                |
    |                                no `Deserialize` in `serialization`
    |
    = help: consider importing this unresolved item through its public re-export instead:
            crate::keys::Deserialize
note: found an item that was configured out
   --> frost-core/src/serialization.rs:171:18
    |
171 | pub(crate) trait Deserialize<C: Ciphersuite> {
    |                  ^^^^^^^^^^^
    = note: the item is gated behind the `serialization` feature
    = help: consider importing this unresolved item through its public re-export instead:
            crate::keys::Serialize
note: found an item that was configured out
   --> frost-core/src/serialization.rs:165:18
    |
165 | pub(crate) trait Serialize<C: Ciphersuite> {
    |                  ^^^^^^^^^
    = note: the item is gated behind the `serialization` feature

error[E0432]: unresolved imports `crate::serialization::Deserialize`, `crate::serialization::Serialize`
   --> frost-core/src/keys/dkg.rs:170:32
    |
170 |     use crate::serialization::{Deserialize, Serialize};
    |                                ^^^^^^^^^^^  ^^^^^^^^^ no `Serialize` in `serialization`
    |                                |
    |                                no `Deserialize` in `serialization`
    |
    = help: consider importing this unresolved item through its public re-export instead:
            crate::keys::Deserialize
note: found an item that was configured out
   --> frost-core/src/serialization.rs:171:18
    |
171 | pub(crate) trait Deserialize<C: Ciphersuite> {
    |                  ^^^^^^^^^^^
    = note: the item is gated behind the `serialization` feature
    = help: consider importing this unresolved item through its public re-export instead:
            crate::keys::Serialize
note: found an item that was configured out
   --> frost-core/src/serialization.rs:165:18
    |
165 | pub(crate) trait Serialize<C: Ciphersuite> {
    |                  ^^^^^^^^^
    = note: the item is gated behind the `serialization` feature

error[E0432]: unresolved imports `crate::serialization::Deserialize`, `crate::serialization::Serialize`
   --> frost-core/src/round1.rs:17:21
    |
17  |     serialization::{Deserialize, Serialize},
    |                     ^^^^^^^^^^^  ^^^^^^^^^ no `Serialize` in `serialization`
    |                     |
    |                     no `Deserialize` in `serialization`
    |
note: found an item that was configured out
   --> frost-core/src/serialization.rs:171:18
    |
171 | pub(crate) trait Deserialize<C: Ciphersuite> {
    |                  ^^^^^^^^^^^
    = note: the item is gated behind the `serialization` feature
note: found an item that was configured out
   --> frost-core/src/serialization.rs:165:18
    |
165 | pub(crate) trait Serialize<C: Ciphersuite> {
    |                  ^^^^^^^^^
    = note: the item is gated behind the `serialization` feature

error: cannot find attribute `serde` in this scope
   --> frost-core/src/lib.rs:156:7
    |
156 |     #[serde(skip)]
    |       ^^^^^

warning: unused imports: `Error`, `Field`, `Group`
 --> frost-core/src/serialization.rs:3:26
  |
3 | use crate::{Ciphersuite, Error, Field, Group};
  |                          ^^^^^  ^^^^^  ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0432`.
warning: `frost-core` (lib) generated 1 warning
error: could not compile `frost-core` (lib) due to 5 previous errors; 1 warning emitted
```

</p>
</details> 

It can be also reproduced by adding `frost-core` as dependency without default features:
```toml
frost-core = { version = "1", default-features = false }
```

--- a/frost-core/CHANGELOG.md
+++ b/frost-core/CHANGELOG.md
@@ -4,6 +4,12 @@ Entries are listed in reverse chronological order.
 
 ## Unreleased
 
+## 1.0.1
+
+* Fixed `no-default-features`, previously it wouldn't compile.
+* Fixed some feature handling that would include unneeded dependencies in some
+  cases.
+
 ## Released
 
 ## 1.0.0

--- a/frost-core/Cargo.toml
+++ b/frost-core/Cargo.toml
@@ -62,7 +62,7 @@ internals = []
 serde = ["dep:serde", "dep:serdect"]
 serialization = ["serde", "dep:postcard"]
 # Exposes ciphersuite-generic tests for other crates to use
-test-impl = ["proptest", "serde_json", "criterion"]
+test-impl = ["dep:proptest", "dep:serde_json", "dep:criterion"]
 # Enable cheater detection
 cheater-detection = []
 

--- a/frost-core/src/keys.rs
+++ b/frost-core/src/keys.rs
@@ -17,13 +17,14 @@ use rand_core::{CryptoRng, RngCore};
 use zeroize::{DefaultIsZeroes, Zeroize};
 
 use crate::{
-    serialization::{Deserialize, Serialize},
-    Ciphersuite, Element, Error, Field, Group, Header, Identifier, Scalar, SigningKey,
-    VerifyingKey,
+    Ciphersuite, Element, Error, Field, Group, Header, Identifier, Scalar, SigningKey, VerifyingKey,
 };
 
 #[cfg(feature = "serde")]
 use crate::serialization::{ElementSerialization, ScalarSerialization};
+
+#[cfg(feature = "serialization")]
+use crate::serialization::{Deserialize, Serialize};
 
 use super::compute_lagrange_coefficient;
 

--- a/frost-core/src/keys/dkg.rs
+++ b/frost-core/src/keys/dkg.rs
@@ -50,6 +50,7 @@ pub mod round1 {
     use derive_getters::Getters;
     use zeroize::Zeroize;
 
+    #[cfg(feature = "serialization")]
     use crate::serialization::{Deserialize, Serialize};
 
     use super::*;
@@ -167,6 +168,7 @@ pub mod round2 {
     use derive_getters::Getters;
     use zeroize::Zeroize;
 
+    #[cfg(feature = "serialization")]
     use crate::serialization::{Deserialize, Serialize};
 
     use super::*;

--- a/frost-core/src/lib.rs
+++ b/frost-core/src/lib.rs
@@ -153,7 +153,7 @@ struct Header<C: Ciphersuite> {
         serde(deserialize_with = "crate::serialization::ciphersuite_deserialize::<_, C>")
     )]
     ciphersuite: (),
-    #[serde(skip)]
+    #[cfg_attr(feature = "serde", serde(skip))]
     phantom: PhantomData<C>,
 }
 

--- a/frost-core/src/round1.rs
+++ b/frost-core/src/round1.rs
@@ -12,14 +12,13 @@ use hex::FromHex;
 use rand_core::{CryptoRng, RngCore};
 use zeroize::Zeroize;
 
-use crate as frost;
-use crate::{
-    serialization::{Deserialize, Serialize},
-    Ciphersuite, Element, Error, Field, Group, Header, Scalar,
-};
+use crate::{Ciphersuite, Element, Error, Field, Group, Header, Scalar};
 
 #[cfg(feature = "serde")]
 use crate::serialization::{ElementSerialization, ScalarSerialization};
+
+#[cfg(feature = "serialization")]
+use crate::serialization::{Deserialize, Serialize};
 
 use super::{keys::SigningShare, Identifier};
 
@@ -353,11 +352,12 @@ where
     /// Computes the [signature commitment share] from these round one signing commitments.
     ///
     /// [signature commitment share]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-14.html#name-signature-share-verificatio
+    #[cfg(any(feature = "cheater-detection", feature = "internals"))]
     #[cfg_attr(feature = "internals", visibility::make(pub))]
     #[cfg_attr(docsrs, doc(cfg(feature = "internals")))]
     pub(super) fn to_group_commitment_share(
         self,
-        binding_factor: &frost::BindingFactor<C>,
+        binding_factor: &crate::BindingFactor<C>,
     ) -> GroupCommitmentShare<C> {
         GroupCommitmentShare::<C>(self.hiding.0 + (self.binding.0 * binding_factor.0))
     }

--- a/frost-core/src/round2.rs
+++ b/frost-core/src/round2.rs
@@ -81,6 +81,7 @@ where
     /// This is the final step of [`verify_signature_share`] from the spec.
     ///
     /// [`verify_signature_share`]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-14.html#name-signature-share-verificatio
+    #[cfg(any(feature = "cheater-detection", feature = "internals"))]
     #[cfg_attr(feature = "internals", visibility::make(pub))]
     #[cfg_attr(docsrs, doc(cfg(feature = "internals")))]
     pub(crate) fn verify(

--- a/frost-core/src/serialization.rs
+++ b/frost-core/src/serialization.rs
@@ -1,6 +1,10 @@
 //! Serialization support.
 
-use crate::{Ciphersuite, Error, Field, Group};
+#[cfg(feature = "serde")]
+use crate::{Ciphersuite, Field, Group};
+
+#[cfg(feature = "serialization")]
+use crate::Error;
 
 #[cfg(feature = "serde")]
 #[cfg_attr(feature = "internals", visibility::make(pub))]
@@ -89,6 +93,7 @@ where
 
 // The short 4-byte ID. Derived as the CRC-32 of the UTF-8
 // encoded ID in big endian format.
+#[cfg(feature = "serde")]
 const fn short_id<C>() -> [u8; 4]
 where
     C: Ciphersuite,

--- a/frost-core/src/tests/ciphersuite_generic.rs
+++ b/frost-core/src/tests/ciphersuite_generic.rs
@@ -326,6 +326,7 @@ fn check_aggregate_errors<C: Ciphersuite + PartialEq>(
     );
 }
 
+#[cfg(feature = "cheater-detection")]
 fn check_aggregate_corrupted_share<C: Ciphersuite + PartialEq>(
     signing_package: frost::SigningPackage<C>,
     mut signature_shares: BTreeMap<frost::Identifier<C>, frost::round2::SignatureShare<C>>,

--- a/frost-ed25519/Cargo.toml
+++ b/frost-ed25519/Cargo.toml
@@ -46,14 +46,15 @@ serde_json = "1.0"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 ## Enable cheater detection
-cheater-detection = ["frost-core/cheater-detection"]
+cheater-detection = ["frost-core/cheater-detection", "frost-rerandomized/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization", "frost-rerandomized/serialization"]
 
 [lib]
 # Disables non-criterion benchmark which is not used; prevents errors

--- a/frost-ed448/Cargo.toml
+++ b/frost-ed448/Cargo.toml
@@ -44,14 +44,15 @@ serde_json = "1.0"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 ## Enable cheater detection
-cheater-detection = ["frost-core/cheater-detection"]
+cheater-detection = ["frost-core/cheater-detection", "frost-rerandomized/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization", "frost-rerandomized/serialization"]
 
 [lib]
 # Disables non-criterion benchmark which is not used; prevents errors

--- a/frost-p256/Cargo.toml
+++ b/frost-p256/Cargo.toml
@@ -45,14 +45,15 @@ serde_json = "1.0"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 ## Enable cheater detection
-cheater-detection = ["frost-core/cheater-detection"]
+cheater-detection = ["frost-core/cheater-detection", "frost-rerandomized/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization", "frost-rerandomized/serialization"]
 
 [lib]
 # Disables non-criterion benchmark which is not used; prevents errors

--- a/frost-rerandomized/Cargo.toml
+++ b/frost-rerandomized/Cargo.toml
@@ -30,13 +30,14 @@ rand_core = "0.6"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 # Exposes ciphersuite-generic tests for other crates to use
-test-impl = ["frost-core/test-impl"]
+test-impl = ["frost-core/test-impl", "serialization"]
 ## Enable cheater detection
 cheater-detection = ["frost-core/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization"]

--- a/frost-ristretto255/Cargo.toml
+++ b/frost-ristretto255/Cargo.toml
@@ -42,14 +42,15 @@ serde_json = "1.0"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 ## Enable cheater detection
-cheater-detection = ["frost-core/cheater-detection"]
+cheater-detection = ["frost-core/cheater-detection", "frost-rerandomized/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization", "frost-rerandomized/serialization"]
 
 [lib]
 # Disables non-criterion benchmark which is not used; prevents errors

--- a/frost-secp256k1/Cargo.toml
+++ b/frost-secp256k1/Cargo.toml
@@ -44,14 +44,15 @@ serde_json = "1.0"
 [features]
 nightly = []
 default = ["serialization", "cheater-detection"]
-serialization = ["serde", "frost-core/serialization"]
 #! ## Features
 ## Enable `serde` support for types that need to be communicated. You
 ## can use `serde` to serialize structs with any encoder that supports
 ## `serde` (e.g. JSON with `serde_json`).
 serde = ["frost-core/serde"]
 ## Enable cheater detection
-cheater-detection = ["frost-core/cheater-detection"]
+cheater-detection = ["frost-core/cheater-detection", "frost-rerandomized/cheater-detection"]
+## Enable a default serialization format. Enables `serde`.
+serialization = ["serde", "frost-core/serialization", "frost-rerandomized/serialization"]
 
 [lib]
 # Disables non-criterion benchmark which is not used; prevents errors

